### PR TITLE
feat(extra/natives): Add GetVehicleWheelSurfaceMaterial

### PR DESCRIPTION
**Pull request by nikez**

### CODE REVIEW

**Comment by blattersturm** [comment]
Simple.

Squeaky.

Clean.

Perfect getter 💯.

LGTM.

**Comment by nikez** [comment]
Added support for 2060 too.

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 1]
```diff
@@ -572,6 +574,11 @@ static HookFunction initFunction([]()
 		context.SetResult<float>(*reinterpret_cast<float*>(wheelAddr + WheelHealthOffset));
 	}));
 
+	fx::ScriptEngine::RegisterNativeHandler("GET_VEHICLE_WHEEL_SURFACE_MATERIAL", makeWheelFunction([](fx::ScriptContext& context, fwEntity* vehicle, uintptr_t wheelAddr)
+	{
+		context.SetResult<unsigned char>(*reinterpret_cast<unsigned char*>(wheelAddr + (Is2060() ? WheelSurfaceMaterialOffset_2060 : WheelSurfaceMaterialOffset_1604)));
```
Comment (blattersturm) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:579]: uh

you know

you can just change the int-thing on init

it's not const

not constexpr

just do it

**Review by nikez — COMMENTED** [review]
Added support for 2060

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp (hunk 1 above)
Comment (nikez) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp:579]: I got tricked... I had it that way :[

Comment (blattersturm) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp:579]: bonus points if you just use the pattern and don't have an initial value

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 2]
```diff
@@ -209,6 +209,7 @@ static int WheelTractionVectorLengthOffset = 0x1B8;
 static int WheelSteeringAngleOffset = 0x1CC;
 static int WheelBrakePressureOffset = 0x1D0;
 static int WheelHealthOffset = 0x1E8; // 75 24 F3 0F 10 81 ? ? ? ? F3 0F
+static int WheelSurfaceMaterialOffset = (Is2060() ? 0x20A : 0x1FE); // 2060 48 8B 4A 10 0F 28 CF F3 0F 59 05 ? ? ? ? - 0x4
```
Comment (blattersturm) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:212]: honestly by now just get yourself the bonus points, this is not even guaranteed to work safely

Comment (nikez) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp]: give me the bonus points now thanks

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 3]
```diff
@@ -313,6 +314,7 @@ static HookFunction initFunction([]()
 		DrawnWheelAngleMultOffset = *hook::get_pattern<uint32_t>("48 8B C8 E8 ? ? ? ? 84 C0 74 ? F3 0F 10 83", 33);
 		VehicleTopSpeedModifierPtr = hook::get_pattern<char>("48 8B D9 48 81 C1 ? ? ? ? 48 89 5C 24 28 44 0F 29 40 C8");
 		VehicleCheatPowerIncreaseOffset = *hook::get_pattern<uint32_t>("E8 ? ? ? ? 8B 83 ? ? ? ? C7 83", 23);
+		WheelSurfaceMaterialOffset = *hook::get_pattern<uint32_t>("48 8B 4A 10 0F 28 CF F3 0F 59 05 ? ? ? ?", -4);
```
Comment (Disquse) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:317]: You can safely remote trailing "? ? ? ?"

**Comment by blattersturm** [comment]
Applied in d5eb603506d94f8d2f58023a34a34aeb16bca02a.

**Pull request closed without merging**